### PR TITLE
Add TaskRepository, convert TaskItem to class.

### LINKED ISSUE #19: Create TaskRepository : ITaskRepository
### Notice
If you want to do this issue, assign yourself to the issue so that we don't have three people submitting PRs for the same issue please :)

### Background / Details
Based on the discussion in #9 and the interface created by #18, create a TaskRepository class that implements the `ITaskRepository` interface, storing the items in memory and saving/reading to file as needed.

### User story
As a user,
I want my tasks to be persisted
so that I can use and solve tasks between sessions.

### Acceptance criteria
- A new class that implements the `ITaskRepository` interface.
- Stores tasks in a file between executions of the program.
- Reads from the file as needed.

If you can think of other things that adds value, feel free to add it.

### PR DESCRIPTION
**Pull request by denislavh**
Closes #19

### CODE REVIEW

File: BusyList/TaskItem.cs [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+{
+    public class TaskItem
+    {
+        public TaskItem(int id, string description, TaskStatus status = TaskStatus.NotStarted)
+        {
+            Id = id;
+            Description = description;
+            TaskStatus = status;
+        }
+        public int Id { get; }
```
Comment (18-F-cali) [inline BusyList/TaskItem.cs:13]: If we implement relative ids won't this be changing everytime.

File: BusyList/TaskItem.cs [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+
+        public string Description { get; }
```
Comment (18-F-cali) [inline BusyList/TaskItem.cs:15]: if the user makes a spelling mistake, he won't be able to change the description

File: BusyList/TaskItem.cs (hunk 1 above)
Comment (pobiega) [reply BusyList/TaskItem.cs:13]: Yes, but we have not decided on relative IDs and even if we did, that was a post 1.0 feature.

File: BusyList/TaskItem.cs (hunk 2 above)
Comment (pobiega) [reply BusyList/TaskItem.cs:15]: We don't have an edit command at this point anyways, so its a moot point.

File: BusyList/TaskRepository.cs [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+        }
+
+        public TaskItem UpdateTask(TaskItem task)
+        {
+
+            Save();
+
+            return task;
+        }
+        private void Save()
```
Comment (18-F-cali) [inline BusyList/TaskRepository.cs:65]: we could add an append to file method insated of  cnverting entire task list everytime so we could call that in 
AddTask() ,  DeleteTask() , UpdateTask().
or we could call this method before application is closed

File: BusyList/TaskRepository.cs [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+        }
+
+        public IEnumerable<TaskItem> GetAll()
+        {
+            return _items;
+        }
+
+        public TaskItem GetTaskById(int id)
+        {
+            return _items.FirstOrDefault(x => x.Id == id);
```
Comment (18-F-cali) [inline BusyList/TaskRepository.cs:55]: should we use SingleOrDefault , as our id will be unique

File: BusyList/TaskItem.cs (hunk 2 above)
Comment (18-F-cali) [reply BusyList/TaskItem.cs:15]: should we set this as init

File: BusyList/TaskRepository.cs (hunk 3 above)
Comment (pobiega) [reply BusyList/TaskRepository.cs:65]: That complicates the writing process massively. For now, lets keep it simple.

**Review by Barraza44 — COMMENTED** [review]
Delete handler class already prints something really similar. Shouldn't just one of them do the printing?

File: BusyList/TaskRepository.cs [hunk 5]
```diff
@@ -0,0 +1,72 @@
...
+            _items.Add(item);
+
+            Save();
+
+            return item;
+        }
+
+        public void DeleteTask(TaskItem task)
+        {
+            Console.WriteLine($"Removed task #{task.Id}");
```
[the remark this answers is not in the document]
Comment (pobiega) [reply BusyList/TaskRepository.cs:41]: Indeed, the task repo itself should not do any form of printing.

**Comment by pobiega** [comment]
Any progress? We are so close to being able to merge this..

**Review by pobiega — APPROVED** [review]

**Review by Barraza44 — APPROVED** [review]

**Pull request merged**